clients/src/main/java/org/apache/kafka/clients/consumer/ConsumerRecord.java

```diff
@@ -105,9 +114,31 @@ public TimestampType timestampType() {
         return timestampType;
     }
 
+    /**
+     * The checksum (CRC32) of the record.
+     */
+    public long checksum() {
+        return this.checksum;
+    }
+
+    /**
+     * The size of the serialized, uncompressed key in bytes.
+     */
+    public int keySize() {
```

Comment:
Would it be better to rename this to serializedKeySize and serializedValueSize to make it clear? If we do this, we probably want to make a similar change in the producer side to make it consistent.